encoder.go

```diff
@@ -157,10 +188,18 @@ func isValueNil(value reflect.Value) bool {
 }
 
 func (encoder *encoder) encode(value reflect.Value, obj *bindings.WAFObject, depth int) error {
-	if encoder.timer.Exhausted() {
+	if encoder.Timer.Exhausted() {
 		return waferrors.ErrTimeout
 	}
 
+	if value.IsValid() && value.CanInterface() {
```

Comment:
`Value.CanInterface()` "returns whether `Interface` can be called without panicing", implying (and the implementations verify this) that `Value.IsValid()`, so there is no need to check both:


```suggestion
	if value.CanInterface() {
```